azuredevops/internal/service/git/resource_git_repository_branch.go

```diff
@@ -0,0 +1,265 @@
+package git
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/git"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/tfhelper"
+)
+
+const (
+	REF_BRANCH_PREFIX = "refs/heads/"
+	REF_TAG_PREFIX    = "refs/tags/"
+)
+
+// ResourceGitRepositoryBranch schema to manage the lifecycle of a git repository branch
+func ResourceGitRepositoryBranch() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceGitRepositoryBranchCreate,
+		ReadContext:   resourceGitRepositoryBranchRead,
+		DeleteContext: resourceGitRepositoryBranchDelete,
+		Importer: &schema.ResourceImporter{
```

Comment:
Get refs API only have the object ID, import this resource may always force a new resource to be created, Recommended that this resource does not support importing.

Reply:
Probably we can add better import that also supports setting `ref_tag`, `ref_branch` or `ref_commit_id` manually using import expression. Something like this is also done in `github_branch`:

<repository_id>:<name>:<ref_(commit_id | tag | branch)>

Not sure if necessary. I'll remove import for now I think.